core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -499,6 +499,12 @@ class SnappyParser(session: SnappySession)
   }
 
   protected final def relationFactor: Rule1[LogicalPlan] = rule {
+    identifier ~ ws ~ '(' ~ ws ~ (expression * commaSep) ~ ws ~ ')' ~>
+        ((fnname: String, e: Any) =>
+          e match {
+            case ve: Vector[Expression] => UnresolvedTableValuedFunction(fnname.toLowerCase(), ve)
+            case _ => UnresolvedTableValuedFunction(fnname, Seq.empty)
+          }) |
```

Comment:
I had it below first. But the parser takes range as unresolved relation and then comes out of the relationFactor completely and then expects group by etc. or where clause and then fails complaining -- Invalid input "(0". To remove this ambiguity in parsing I moved it to top.